spec.go

```diff
@@ -0,0 +1,9 @@
+package valuemodifier
+
+// ValueModifier implements some modification mechanism for values being
+// provided. This can e.g. be an implementation to encrypt a given value using
+// GPG encryption standards or encode a given value using base64 encoding
+// standard.
+type ValueModifier interface {
+	Modify(value []byte) ([]byte, error)
+}
```

Comment:
The idea of the value modifier is to provide some configurable mechanism to modify values of arbitrary structures. In our concrete case we want to encrypt/decrypt credentials in config files. Before we processed the whole file and got an ugly blob of some GPG message. This was not really maintainable. The value modifiers help us only processing the credential values by keeping the structure of config files. See the tests how this works in detail. Note that this serves my preparations for the host cluster discovery. In the end I want to have a repository that contains all necessary information about an installation. Our tools provide then functionality to list host clusters and other information.